test/Benchmarks/src/Vector/Utils.enso

```diff
@@ -4,5 +4,48 @@ polyglot java import java.util.Random as Java_Random
 
 make_random_vec : Integer -> Vector
 make_random_vec n =
-    random_gen = Java_Random.new n
-    Vector.fill n random_gen.nextDouble
+    random_gen = Java_Random.new
+    Vector.new n (_-> random_gen.nextDouble)
```

Comment:
@radeusgd, have you tried to inspect the generated vector? It is full of equal values! Using `fill` isn't a good idea to get different elements in the vector.

Also using `Java_Random.new n` doesn't really generate _random vector_, but rather always the same sequence. Is that desired?

Reply:
> @radeusgd, have you tried to inspect the generated vector? It is full of equal values! Using `fill` isn't a good idea to get different elements in the vector.

Interesting, I think in the past `fill` used to re-run side effects for each element, so this used to work. It must have been changed sometime. I agree `fill` was not the best idea to begin with, I'm not sure why we went for it.

> Also using `Java_Random.new n` doesn't really generate _random vector_, but rather always the same sequence. Is that desired?

Yes, I think we wanted this to be deterministic. The sequence should be 'random' but we wanted the same sequence in each run - to avoid benchmark instability due to some outliers in the sequence I guess. Is there any benefit to randomizing the sequence between each runs? IMO it is worse because we lose reproducibility.

Reply:
OK. Returning back the determinism: 1e9efd7

Reply:
> in the past `fill` used to re-run side effects for each element

[Vector.fill delegates to new](https://github.com/enso-org/enso/blame/5fad3558a6858864d7d5c43e40cd98f5b1496fd4/distribution/lib/Standard/Base/0.0.0-dev/src/Data/Vector.enso#L149) since 
- #3744

I guess the change in behavior comes from there as the [PR removed `~` from `item` argument](https://github.com/enso-org/enso/pull/3744/files#r1627030213).